### PR TITLE
Update to Bootstrap 5

### LINKED ISSUE #2513: Update to Bootstrap 5
Work in progress here: https://github.com/sul-dlss/exhibits/compare/bs5?expand=1

Depends on:
- [x] New release of Spotlight with https://github.com/projectblacklight/spotlight/commit/b0873d3a9bc0f140682cc6cad5eac0ca6eaa53f6
- [x] Updating bootstrap_form to `~> 5.4` in Exhibits

Gotchas: blacklight_range_limit does not work with Bootstrap 5:
* https://github.com/projectblacklight/blacklight_range_limit/issues/177
* https://github.com/projectblacklight/blacklight_range_limit/issues/210
* https://github.com/projectblacklight/blacklight_range_limit/issues/212
* https://github.com/projectblacklight/blacklight_range_limit/issues/231 

There are accessibility concerns with blacklight_range_limit anyway, and based on discussions with @caaster a more limited date facet UI may be acceptable. Here's what we did for stanford-arclight for now:
https://github.com/sul-dlss/stanford-arclight/commit/c9264c80152e388f599711a2663e18636e12c81c

TODO:

- [x] Search tips modal has an empty button on the lower right
- [x] More details modal has 3 scroll bars
- [x] More details layout and lower button bar layout needs adjustment
- [x] BL Range Limit: turn off JS or update to new version once released
- [x] SUL logo has visible text over logo
- [x] IIIF logo should to the lower right side of the thumbnail (not lower left)
- [x] BL gallery selected layout is not displayed as active
- [x] Check that active style is applied to grouped by collection results
- [x] Tags are displaying as buttons and not links
- [x] Search bar is missing for "Find items in all exhibits"
- [x] Drop down next to site wide search should not have rounded corners on the right side
- [x] Exhibit search field select dropdown is too narrow
- [x] Test Parker specific features
- [x] Parker more details modal has an extra scrollbar
- [x] exhibit search without a search field dropdown should have rounded corners on left side. (this appears to be a blacklight bug: https://github.com/projectblacklight/blacklight/blob/main/app/components/blacklight/search_bar_component.html.erb#L25-L30 )
- [x] Page width is off -- scrolls off on the right side
- [x] indexing status modal has horizontal scrollbar
- [x] Top nav & bulk actions menus don't work
- [x] browse category search bar and button should line up
- [x] browse category search bar should not look terrible when there's a search term present

### LINKED ISSUE #2641: Parker exhibit should now have the same exhibits footer as other exhibits
Per agreement with Ben Albritton, the Parker exhibit (https://exhibits.stanford.edu/parker) can now have the same footer as all other Spotlight at Stanford exhibits.

### LINKED ISSUE #2654: Exhibits thumbnail redesign
Exhibits thumbnail
- Background color: **Fog Light** `#F4F4F4` Opacity: 90%

Exhibits category (All, Art & Humanities, Social Justice, etc.)
- Background color: **Fog Light** `#F4F4F4` Opacity: 100% 
- Font weight : `600` for active selection

https://www.figma.com/design/0LTbMiP5xdL6T94nmlKDci/Exhibits-Redesign-2024?node-id=465-10235&t=WZb88CX43UyeiIJp-4
Image

### PR DESCRIPTION
**Pull request by corylown**
Known changes:
- Footer updated to use component library
- Link styles (impacts underlining and color) updated to use component library
- Facet constraint controls updated to use component library
- The date range facet now does not display the date slider control or the histogram because these features are not compatible with Bootstrap 5.

Known issues:
- For exhibits that only have one search field configured the search box has squared off edges on the left side. This is a Blacklight bug that I have submitted a PR to fix. The PR has been accepted to Blacklight but is not in a release yet.
- ~~The browse category search box styling is broken. The X to clear the current search breaks out of the search box on Bootstrap 5. This is a Spotlight issue, but it's very difficult to get this to work right for both Bootstrap 4 & 5. Trying to come up with a good solution.~~ This is fixed now.

closes #2654
closes #2513 
closes #2641

### CODE REVIEW

**Comment by taylor-steve** [comment]
Looking great! As always, here are some differences presented mostly without comment. Apologies if some are intentionally different.


* Possible issue with widgets configured for right positioned text: https://exhibits-stage.stanford.edu/lanciani
Screenshot 2024-11-21 at 9 47 05 AM

* Left text not wrapping?
Stage (https://exhibits-stage.stanford.edu/american-enlightenment/feature/the-lewis-and-clark-expedition): 
Screenshot 2024-11-21 at 10 00 54 AM
Prod:
Screenshot 2024-11-21 at 10 01 03 AM

* Analytics grid styling (also some missing underlines on that page, but IDK if those were even intentional)

Screenshot 2024-11-21 at 10 07 03 AM

* Missing spacing on feedback recipients from General config:
Screenshot 2024-11-21 at 10 09 19 AM

* Private tag text color

Stage:
Screenshot 2024-11-21 at 10 11 35 AM

Prod:
Screenshot 2024-11-21 at 10 11 45 AM

**Comment by corylown** [comment]
Also seeing underlining when the exhibit card text/link wraps:

Screenshot 2024-11-21 at 3 15 13 PM

**Comment by caaster** [comment]
Analytics grid styling issues, as Steve noted. 

Specifically the following items:

1. The outline is only on the bottom and not on the top so the grids float weirdly (need the top outline to visually anchor the boxes/rows:
Screen Shot 2024-11-22 at 3 12 43 PM

2. The line underneath "Most popular pages" and "Most used search terms" is missing, which scrunches the info together on the page, making it harder to read.

**Comment by caaster** [comment]
I know we've discussed this before, so maybe a fix just hasn't been deployed? For list view, the MD under each item title is way too far to the right:

Screen Shot 2024-11-22 at 3 21 14 PM

**Comment by caaster** [comment]
There is a radical difference in the size displayed for the first image on the home page in this exhibit. I believe they are the same image, and they use the same widget, and I made sure I am using the same 100% screen resolution in both -stage and -prod browser tabs.

Here is the home page in stage:
Screen Shot 2024-11-22 at 3 25 02 PM

Here is the same home page in prod:
Screen Shot 2024-11-22 at 3 27 26 PM

**Comment by caaster** [comment]
There seems to be a significant difference in visual page display (especially but not only for the text) for stage vs prod. Is it the kerning? Switch back and forth a few times between these 2 pages and you will see what I mean:

Stage: https://exhibits-stage.stanford.edu/preventing-genocide
Prod: https://exhibits.stanford.edu/preventing-genocide

File: app/assets/stylesheets/sul_variables.scss [hunk 1]
```diff
@@ -18,6 +18,7 @@ $stone: #544948;
 $very-pale-orange: #ffe9c5;
 $very-pale-red: #ffe5ea;
 $white: #fff;
+$fog-light: #f4f4f4;
```
Comment (dnoneill) [inline app/assets/stylesheets/sul_variables.scss:21]:
```suggestion
$fog-light-90: #f4f4f4e6;
```

File: app/assets/stylesheets/variables.scss [hunk 2]
```diff
@@ -20,8 +20,8 @@ $color-gray-80: #ccc;
 
 $secondary: $color-beige-10;
 
-$exhibit-card-bg: $color-beige-20;
-$exhibit-card-border: $color-gray-80;
+$exhibit-card-bg: $fog-light;
+$exhibit-card-border: $fog-light;
```
Comment (dnoneill) [inline app/assets/stylesheets/variables.scss:24]:
```suggestion
$exhibit-card-bg: $fog-light-90;
$exhibit-card-border: $fog-light-90;
```

Comment (dnoneill) [inline app/assets/stylesheets/variables.scss:24]:
```suggestion
$exhibit-card-bg: $fog-light-90;
$exhibit-card-border: $fog-light-90;
```

**Comment by taylor-steve** [comment]
Flagging more potential areas after stage was updated:

* Search tips modal close button:
Screenshot 2024-11-26 at 10 17 58 AM

* Mobile sized limit your search/facets border & underline
Screenshot 2024-11-26 at 10 19 01 AM

* Create new exhibit labels:
Screenshot 2024-11-26 at 10 30 21 AM

**Comment by corylown** [comment]
Thanks @taylor-steve

- I'd prefer to handle the search tips modal close button alignment as a separate issue, because I think the markup of that modal is...not quite right and may need a refactor beyond what I want to tackle here. I'll create a separate issue for that.
- I removed the underline beneath "Limit your search" because it's not part of the new component library and we were adding it in exhibits.
- Now, all the field labels are bold and right aligned (which is consistent with exhibits prod and I don't know what forms are supposed to look like since they're not in the component library yet).

**Comment by corylown** [comment]
Here's the issue for the search tips modal: https://github.com/sul-dlss/exhibits/issues/2657

**Comment by corylown** [comment]
@dnoneill I'm pretty sure that's the link underline style from the component library

**Comment by dnoneill** [comment]
It is. I think we can ignore.

**Comment by dnoneill** [comment]
Admin dropdown should be right aligned on the dropdown button corner. This is causing it to get cut off on medium size pages.

Stage:
Screenshot 2024-11-26 at 3 57 39 PM

Production:
Screenshot 2024-11-26 at 3 57 43 PM

**Comment by corylown** [comment]
@dnoneill good find -- that's because `dropdown-menu-right` is `dropdown-menu-end` in BS5

File: app/components/search_tips_component.html.erb [hunk 3]
```diff
@@ -5,18 +5,18 @@
   <% end %>
   <% component.with_body do %>
     <ul class="mt-3 pr-4 pb-2">
```
Comment (dnoneill) [inline app/components/search_tips_component.html.erb:7]:
```suggestion
    <ul class="mt-3 pr-4 pb-2">
    <ul class="mt-3 pe-4 pb-2">
```

Comment (dnoneill) [reply app/components/search_tips_component.html.erb:7]: padding right becomes padding end in bootstrap 5

File: app/components/search_tips_component.html.erb [hunk 4]
```diff
@@ -5,18 +5,18 @@
...
+      <li> Use "OR", "AND", and "NOT" (uppercase) to create <span class="fw-bold">Boolean</span> logic within a field. You can use parentheses in your complex expressions. </li>
+      <li> Use "*" to search a <span class="fw-bold">truncated</span> term. (Word-stemming is done automatically.) </li>
+      <li> Use "?" as a <span class="fw-bold">wildcard</span> to replace a character in a word. </li>
+      <li> <span class="fw-bold">Word-stemming</span> automatically includes plural and singular forms, and common suffix and tense variations. </li>
     </ul>
   <% end %>
   <% component.with_footer do %>
-    <button type="button" class="btn btn-outline-primary" data-bs-dismiss="modal" data-dismiss="modal">
-      <span aria-hidden="true" class="visually-hidden"><%= t('search_tips.close') %> </span>
+    <button type="button" class="btn btn-outline-primary" data-bl-dismiss="modal" data-dismiss="modal">
```
Comment (dnoneill) [inline app/components/search_tips_component.html.erb:18]:
```suggestion
    <button type="button" class="btn btn-outline-primary" data-bl-dismiss="modal" data-dismiss="modal">
    <button type="button" class="btn btn-outline-primary" data-bs-dismiss="modal" data-bl-dismiss="modal">
```

data-dismiss doesn't do anything I think.

**Comment by dnoneill** [comment]
Honestly, I am not sure we actually want to fix this because I think it kind of looks better on stage, but I figured it should be noted. Button placement is different.

Stage
Screenshot 2024-11-26 at 5 15 04 PM

Prod
Screenshot 2024-11-26 at 5 15 36 PM

**Comment by dnoneill** [comment]
Image cropping not working: https://exhibits-stage.stanford.edu/california-as-an-island/appearance/edit

**Comment by dnoneill** [comment]
https://exhibits-stage.stanford.edu/california-as-an-island/appearance, main menu reordering is erroring with ```Masthead image file size must be less than 10 MB```. I am guessing this parity because the second item in the main menu on prod is curated features but the menu still says history.

**Comment by dnoneill** [comment]
Adding a new user also removes the previous user. https://exhibits-stage.stanford.edu/california-as-an-island/users. There used to be 4 users on this page. I added a new user twice and it deleted a user each time a user was added.

**Comment by corylown** [comment]
I can't duplicate the behavior re: adding a user removes a user.

**Comment by corylown** [comment]
Pretty sure the `image file size must be less than 10 MB` error is a data issue with the exhibit and unrelated to BS5.

**Comment by corylown** [comment]
I'm not worrying about layout changes to forms unless there are inconsistencies or they look bad.

**Comment by dnoneill** [comment]
Pretty minor.

Stage
Screenshot 2024-11-26 at 5 30 50 PM

Prod
Screenshot 2024-11-26 at 5 30 54 PM

**Comment by corylown** [comment]
@dnoneill can you create a spotlight issue for that last one with the upload button? I don't think it's worth holding up this work for, but we should fix it in spotlight when we have time. My guess it's an `input-group-append` that I missed that is no longer a thing in BS5.

**Review by taylor-steve — APPROVED** [review]

**Pull request merged**